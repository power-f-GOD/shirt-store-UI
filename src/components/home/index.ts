export * from './BottomBar';
export * from './Card';
