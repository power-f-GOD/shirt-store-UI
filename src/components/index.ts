export * from './Stack';
export * from './Text';
export * from './View';
export * from './Img';
export * from './Skeleton';
