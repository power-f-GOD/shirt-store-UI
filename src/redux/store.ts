import { configureStore, Dispatch, PayloadAction } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useSelector } from 'react-redux';

import { DEVELOPMENT } from 'src/constants/misc';
import { rootReducer } from './slices';

const store = configureStore({
  reducer: rootReducer,
  devTools: DEVELOPMENT,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false })
});

export const dispatch: Dispatch<PayloadAction<any> | any> = store.dispatch;
export const getState = store.getState;

export type StateProps = ReturnType<typeof getState>;
export type StoreDispatch = ReturnType<typeof dispatch>;
export type StoreGetState = typeof getState;

// export const useTypedDispatch: () => StoreDispatch = useReduxDispatch;
export const useTypedSelector: TypedUseSelectorHook<StateProps> = useSelector;

export default store;
