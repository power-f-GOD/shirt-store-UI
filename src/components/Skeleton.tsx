import { FC, memo, useMemo } from 'react';

import { SkeletonProps } from 'src/types';
import { Stack } from '.';

const _Skeleton: FC<SkeletonProps> = ({
  size,
  width,
  height,
  type,
  borderRadius,
  count,
  variant,
  erred,
  className: _className,
  style: _style
}) => {
  const hasRoundedClass = /(^|\s)rounded/.test(_className || '');
  const hasWidthClass = /(^|\s)w-/.test(_className || '');
  const hasHeightClass = /(^|\s)h-/.test(_className || '');
  const className = `${_className || ''}${
    hasHeightClass || height ? '' : ' h-1m'
  } ${type === 'circle' ? ' rounded-full' : ''}`;
  const style = useMemo(
    () => ({
      ..._style,
      ...((height || size) && !hasHeightClass
        ? { height: height || size }
        : {}),
      ...((width || size) && !hasWidthClass ? { width: width || size } : {}),
      ...(borderRadius && !hasRoundedClass ? { borderRadius } : {})
    }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      // eslint-disable-next-line react-hooks/exhaustive-deps
      JSON.stringify(_style),
      borderRadius,
      hasHeightClass,
      hasRoundedClass,
      hasWidthClass,
      height,
      size,
      width
    ]
  );

  return (
    <>
      {Array(count || 1)
        .fill('')
        .map((_, i) => (
          <Stack
            key={i}
            as="span"
            className={`Skeleton relative left-0 top-0 inline-flex justify-center items-center self-center max-width-full overflow-hidden rounded-sm ${
              className || ''
            } ${variant || ''} ${erred ? 'erred' : ''}`
              .replace(/\s+/g, ' ')
              .trim()}
            style={style}
          />
        ))}
    </>
  );
};

export const Skeleton = memo(_Skeleton);
