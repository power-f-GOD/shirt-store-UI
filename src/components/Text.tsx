import { memo, useMemo, forwardRef } from 'react';

import { TextProps } from 'src/types';
import { View } from './View';

const _Text = forwardRef<HTMLElement, TextProps>(
  (
    {
      as,
      children,
      style,
      className,
      height,
      width,
      background,
      color,
      display,
      fontSize,
      overflow,
      overflowY,
      overflowX,
      alignSelf,
      fontStyle,
      fontWeight,
      overflowEllipsis,
      textAlign,
      textTransform,
      component,
      ...restProps
    },
    ref
  ): JSX.Element => {
    const Component = component || View;
    const styleCheck = JSON.stringify(style || {});

    return (
      <Component
        {...restProps}
        as={as || 'span'}
        ref={ref}
        className={`Text ${className || ''}${
          overflowEllipsis ? ' text-ellipsis' : ''
        }`.trim()}
        style={useMemo(
          () => ({
            alignSelf,
            width,
            height,
            color,
            background,
            display,
            fontSize,
            fontStyle,
            fontWeight,
            overflow,
            overflowY,
            overflowX,
            textAlign,
            textTransform,
            ...(style || {})
          }),
          // eslint-disable-next-line react-hooks/exhaustive-deps
          [
            height,
            width,
            background,
            color,
            display,
            fontSize,
            fontStyle,
            fontWeight,
            overflow,
            overflowX,
            overflowY,
            textAlign,
            textTransform,
            styleCheck
          ]
        )}>
        {children}
      </Component>
    );
  }
);

_Text.displayName = '_Text';

export const Text = memo(_Text);
