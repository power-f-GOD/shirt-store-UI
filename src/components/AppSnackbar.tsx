import React, { useCallback, memo } from 'react';
import { shallowEqual } from 'react-redux';
import {
  Snackbar as MuiSnackbar,
  Button,
  Alert,
  Slide,
  SlideProps,
  AlertTitle
} from '@mui/material';

import { dispatch, snackbar, useTypedSelector } from 'src/redux';

const _AppSnackbar = () => {
  const {
    open,
    severity,
    variant,
    message,
    label,
    duration,
    position,
    title,
    autoHide
  } = useTypedSelector((state) => state.snackbar, shallowEqual);

  const handleOnClose = useCallback(() => {
    dispatch(snackbar({ open: false }));
  }, []);

  const SlideTransition = useCallback(
    (props: SlideProps) => {
      return (
        <Slide
          {...props}
          direction={
            position === 'top' || (!position && severity !== 'info')
              ? 'down'
              : 'up'
          }
        />
      );
    },
    [position, severity]
  );

  return (
    <MuiSnackbar
      open={!!open}
      autoHideDuration={duration || 6000}
      onClose={autoHide ? handleOnClose : undefined}
      action={
        <Button color={severity} size="small" onClick={handleOnClose}>
          {label || 'OK'}
        </Button>
      }
      TransitionComponent={
        // severity === 'info' ? undefined :
        SlideTransition
      }
      anchorOrigin={
        position === 'top' || (!position && severity !== 'info')
          ? { vertical: 'top', horizontal: 'right' }
          : globalThis.innerWidth < 768
          ? { vertical: 'bottom', horizontal: 'right' }
          : undefined
      }>
      <Alert
        onClose={handleOnClose}
        severity={severity}
        variant={variant}
        className="shadow-lg">
        {title && (
          <AlertTitle className="text-capitalize mb-1">
            {typeof title === 'boolean' ? `${severity}!` : title}
          </AlertTitle>
        )}
        {message && <>{message[0].toUpperCase() + message.slice(1)}</>}
      </Alert>
    </MuiSnackbar>
  );
};

export const AppSnackbar = memo(_AppSnackbar);
