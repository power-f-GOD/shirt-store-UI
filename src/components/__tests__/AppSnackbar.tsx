import { render } from '@testing-library/react';
import { Provider as ReduxProvider } from 'react-redux';

import store from 'src/redux/store';
import { AppSnackbar } from 'src/components/AppSnackbar';

it('renders the AppSnackbar component unchanged', () => {
  const { container } = render(
    <ReduxProvider store={store}>
      <AppSnackbar />
    </ReduxProvider>
  );

  expect(container).toMatchSnapshot();
});
