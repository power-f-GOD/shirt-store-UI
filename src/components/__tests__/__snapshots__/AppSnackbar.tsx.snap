// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the AppSnackbar component unchanged 1`] = `<div />`;
